TwilioVerifySDK/TwilioVerify/Sources/Data/Storage+AccessGroup.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  StorageProvider+AccessGroup.swift
+//  TwilioVerifySDK
+//
+//  Copyright © 2020 Twilio.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+extension Storage {
+
+  // MARK: - Constants
+
+  enum MigrationDirection {
+    case toAccessGroup, fromAccessGroup
+  }
+
+  // MARK: - Factors
+
+  /// Returns all the stored factors excluding parameters such as accountService or accessGroup.
+  func getAllFactors(
+    using factorMapper: FactorMapperProtocol
+  ) -> [Factor] {
+    let query: CFDictionary = [
+      kSecClass: kSecClassGenericPassword,
+      kSecReturnAttributes: true,
+      kSecReturnData: true,
+      kSecMatchLimit: kSecMatchLimitAll,
+      kSecAttrAccessible: kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly
+    ] as CFDictionary
+
+    var result: AnyObject?
+
+    guard SecItemCopyMatching(query, &result) == errSecSuccess else {
+      Logger.shared.log(withLevel: .error, message: "Failed to obtain stored Factors.")
+      return []
```

Comment:
Should we throw an error here?

Reply:
After discussing this, we prefer to not throw an error for the migration, but we should an error log and mention this edge case in the README